docs/integration/opensearch-integration.md

```diff
@@ -40,6 +40,135 @@ OpenSearch runs within a Java Virtual Machine (JVM) and [monitoring JVM](https:/
+- Recover unassigned shards and allocate them to available nodes in the cluster
+
+### Thread pool rejections anomaly
+
+This alert rule continuously monitors thread pool rejections in an OpenSearch environment, identifying anomalies in the rate at which thread pool requests are rejected. When anomalies are detected, it triggers a warning (WARN priority). The minimum delay between consecutive alerts triggered by this alert rule is set to 10 minutes.
+
+Suppose an OpenSearch cluster experiences a sudden increase in thread pool rejections, potentially due to resource limitations or high query loads. When this happens, the alert rule checks for anomalies in thread pool rejections over the last 90 minutes. Upon detecting the anomaly, the alert rule triggers a warning.
+
+#### Actions to take
+
+- Review system metrics for the OpenSearch cluster, including CPU, memory, and disk usage, for any resource constraints that may be contributing to thread pool rejections
+- Analyze query patterns for any inefficient or resource-intensive queries. Optimize queries to reduce the load on the cluster
+- Review thread pool configurations and adjust settings such as thread counts, queue sizes, and timeouts to better accommodate the workload and prevent thread pool saturation
+
+### Used memory > 80%
+
+This alert rule continuously monitors memory usage in an OpenSearch environment and triggers a warning (WARN priority) when the used memory exceeds 80% of the total available memory. The minimum delay between consecutive alerts triggered by this alert rule is set to 10 minutes.
+
+Suppose the OpenSearch cluster experiences a sudden increase in activity or data volume, leading to increased memory usage. When the used memory exceeds 80% of the total available memory, the alert rule checks memory usage over the last hour. Upon crossing the threshold, the alert rule triggers a warning.
+
+#### Actions to take
+
+- Investigate the cause of increased memory usage in the OpenSearch cluster, such as processes or activities contributing to the high memory consumption
+- Review and optimize the configuration settings of the OpenSearch cluster, including heap size allocation and cache settings
+
+### Swap usage
+
+This alert rule continuously monitors swap usage in an OpenSearch environment by tracking the rate of swap input/output operations. When swap usage exceeds the specified threshold, it triggers a warning (WARN priority). The minimum delay between consecutive alerts triggered by this alert rule is set to 10 minutes.
+
+Suppose the swap usage on a node in the OpenSearch cluster suddenly spikes, indicating potential resource constraints or performance issues. When this happens, the alert rule checks for swap usage over the last 10 minutes. The alert is triggered when the sum of swap input/output operations within the specified timeframe is more than 1 (the threshold value).
```

Comment:
@radu-gheorghe I think I'm wrong here. Can you confirm the condition for triggering this alert?

Reply:
Actually, I think any swap usage (even if you read 1 byte) would trigger the alert. The idea is that OpenSearch needs OS cache (read: free memory) to work with the index. If the OS swaps the JVM memory, it means one of the following: there's not enough RAM on the box or (more commonly) the OS decides to swap a chunk of RAM preemptively (because of an inappropriate `vm.swappiness` setting. Either way, it could introduce significant (read: unacceptable) delays when OpenSearch wants to access that piece of memory, so it's a no-no to swap.

Reply:
Thanks @radu-gheorghe! I've updated the AR.